File: kvt-projekat2024/src/app/models/Review.ts
import { User } from "./User";
import { Comment } from "./Comment";
import { Facility } from "./Facility";
import { Rate } from "./Rate";

export class Review {
    id: number;
    createdAt: Date;
    exerciseCount: number;
    hidden: boolean;
    isActive: boolean; 
    user: User;
    facilityName: string; // Promenjen naziv
    rate?: {
      equipment: number;
      staff: number;
      hygiene: number;
      space: number;
    };
    commentText?: string;
  
    constructor(
      id: number,
      createdAt: Date = new Date(),
      exerciseCount: number = 0,
      hidden: boolean = false,
      isActive: boolean = true,
      user: User,
      facilityName: string, // Promenjen naziv
      rate: Rate,
      commentText?: string
    ) {
      this.id = id;
      this.createdAt = createdAt;
      this.exerciseCount = exerciseCount;
      this.hidden = hidden;
      this.isActive = isActive;
      this.user = user;
      this.facilityName = facilityName;
      this.rate = rate;
      this.commentText = commentText;
    }
  }

File: svt-projekat2024/src/main/java/rs/ac/uns/ftn/svt/model/Review.java
package rs.ac.uns.ftn.svt.model;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@JsonIgnoreProperties({"user", "facility", "comment", "rate"})
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime createdAt;

    private Integer exerciseCount;

    private Boolean hidden;

    @Column(name = "is_active")
    private Boolean isActive;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "comment_id")
    private Comment comment;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "facility_id", nullable = false)
    private Facility facility;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "rate_id", nullable = false)
    private Rate rate;


}
